src/Gui/NotificationBox.cpp

```diff
@@ -0,0 +1,352 @@
+/***************************************************************************
+ *   Copyright (c) 2023 Abdullah Tahiri <abdullah.tahiri.yo@gmail.com>     *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef _PreComp_
+# include <memory>
+# include <mutex>
+# include <QApplication>
+# include <QAction>
+# include <QActionEvent>
+# include <QDesktopWidget>
+# include <QEvent>
+# include <QHBoxLayout>
+# include <QHeaderView>
+# include <QLabel>
+# include <QMenu>
+# include <QPointer>
+# include <QScreen>
+# include <QStyleOption>
+# include <QStylePainter>
+# include <QStringList>
+# include <QTextDocument>
+# include <QTimer>
+#endif
+
+#include "NotificationBox.h"
+
+using namespace Gui;
+
+namespace Gui {
+
+/** Class showing the notification as a label
+ * */
+class NotificationLabel : public QLabel
+{
+    Q_OBJECT
+public:
+    // Windows implementation uses QWidget w to pass the screen (see NotificationBox::showText).
+    // This screen is used as parent for QLabel.
+    // Linux implementation does not rely on a parent (w = nullptr).
```

Comment:
I can find no reason. It is just how QT programmers did it for QToolTip. 

Before trying to emulate QToolTip, I tried many other solutions. All had issues, either with focus, or they were not fully non-modal or otherwise interfered. For this reason, I have been trying not to change their design decisions. The fact that I cannot test on Windows platform also helps. 

However, I also found it strange, and that is why I have documented it.